pkg/scalers/aws_cloudwatch_scaler.go

```diff
@@ -178,6 +182,24 @@ func parseAwsCloudwatchMetadata(config *ScalerConfig) (*awsCloudwatchMetadata, e
 		}
 	}
 
+	if val, ok := config.TriggerMetadata["metricEndTimeOffset"]; ok && val != "" {
+		metricEndTimeOffset, err := strconv.Atoi(val)
+		if err != nil {
+			cloudwatchLog.Error(err, "Error parsing metricEndTimeOffset metadata")
```

Comment:
I know that this is the existing behavior in other parameters, but is it correct? I mean, I'd like to have an error if the parameter that I set is incorrect instead of using a default value (?). 
@zroubalik ?

Reply:
Yeah you are right, we should probably check the behavior here, to be consistent with the rest of the scalers.

Reply:
@JorTurFer totally agree, I didn't add a check just because I didn't see existing checks for the current inputs. In fact, aws api has a very strict requirement on the inputs. other than the UNIT you have mentioned, the `Stat` must be one of `SampleCount, Average, Sum, Minimum and Maximum`, and for the Period, it cannot be arbitrary values either:

> The granularity, in seconds, of the returned data points. For metrics with regular resolution, a period can be as short as one minute (60 seconds) and must be a multiple of 60. For high-resolution metrics that are collected at intervals of less than one minute, the period can be 1, 5, 10, 30, 60, or any multiple of 60. High-resolution metrics are those metrics stored by a PutMetricData call that includes a StorageResolution of 1 second. 

https://docs.aws.amazon.com/cli/latest/reference/cloudwatch/get-metric-data.html

Reply:
Don't worry, as @zroubalik said, we should check the behavior because it's a bit inconsistent.
But in any case, if you could do it, it would be nice!! Basically behavior in other scalers is returning an error the parameter is not correct. For example these are some lines from [MSSQL Scaler](https://github.com/kedacore/keda/blob/main/pkg/scalers/mssql_scaler.go#L95-L104)
```golang
	// Target query value
	if val, ok := config.TriggerMetadata["targetValue"]; ok {
		targetValue, err := strconv.Atoi(val)
		if err != nil {
			return nil, fmt.Errorf("targetValue parsing error %s", err.Error())
		}
		meta.targetValue = targetValue
	} else {
		return nil, fmt.Errorf("no targetValue given")
	}
```
Every time that happens any error, the error is directly returned

Reply:
@JorTurFer I can do it. actually, I have also seen some other inconsistent behavior in the code:
```
	if val, ok := config.TriggerMetadata["metricCollectionTime"]; ok && val != "" {
		metricCollectionTime, err := strconv.Atoi(val)
		if err != nil {
			cloudwatchLog.Error(err, "Error parsing metricCollectionTime metadata")
		} else {
			meta.metricCollectionTime = int64(metricCollectionTime)
		}
	}


	if val, ok := config.TriggerMetadata["metricStat"]; ok && val != "" {
		meta.metricStat = val
	}


	if val, ok := config.TriggerMetadata["metricStatPeriod"]; ok && val != "" {
		metricStatPeriod, err := strconv.Atoi(val)
		if err != nil {
			cloudwatchLog.Error(err, "Error parsing metricStatPeriod metadata")
		} else {
			meta.metricStatPeriod = int64(metricStatPeriod)
		}
	}
```

in stead of returning  err when it's not nil, this two block just log it instead. I am going to fix them all together

Reply:
I can't mark as resolved, IDK why... :(